### DISCUSSION TITLE
0.34.4 happy-dom based tests failing with  Cannot set properties of undefined (setting 'href')

### DISCUSSION THREAD

**Discussion by EvHaus** [vitest-dev/vitest#4099] [Q&A] [answered]
After upgrading from 0.34.3 to 0.34.4 my happy-dom based React tests are now failing with:

```
TypeError: Cannot set properties of undefined (setting 'href')
 ❯ new Window ../../node_modules/happy-dom/src/window/Window.ts:355:22
 ❯ Object.setupVM ../../node_modules/vitest/dist/vendor-environments.c9c96bf7.js:541:17
 ❯ run ../../node_modules/vitest/dist/vm.js:83:14
 ❯ ../../node_modules/tinypool/dist/esm/worker.js:109:20
```

I'm looking through the changelog to try to see what might have caused it and I'm not sure if this is a bug or if I need to make some changes to my configs. Any pointers?

**Comment by sheremet-va** [comment, marked as answer]
This looks to be a bug in happy-dom (location is initialized after constructor code). I fixed it by updating happy-dom to the latest version.

**Reply by EvHaus** [reply]
Thanks. That was it. I was running a pretty old version of `happy-dom` (7.4.0). Upgrading to 11.0.2 fixed it for me.
